src/java/org/apache/cassandra/schema/SchemaKeyspace.java

```diff
@@ -563,6 +565,11 @@ private static void addTableParamsToRowBuilder(TableParams params, Row.SimpleBui
         // in mixed operation with pre-4.1 versioned node during upgrades.
         if (params.memtable != MemtableParams.DEFAULT)
             builder.add("memtable", params.memtable.configurationKey());
+
+        // As above, only add the allow_auto_snapshot column if the value is not default (true) and
+        // auto-snapshotting is enabled, to avoid RTE in pre-4.2 versioned node during upgrades
+        if (DatabaseDescriptor.isAutoSnapshot() && !params.allowAutoSnapshot)
```

Comment:
I would eliminate the `DatabaseDescriptor.isAutoSnapshot()` condition here - I'm unsure why it was added here? Only the value of the table param is important here.